codec/src/main/java/io/netty/handler/codec/Headers.java

```diff
@@ -638,353 +606,285 @@
-     * {@link java.util.Calendar}, which are formatted to the date format defined in <a
-     * href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3.1">RFC2616</a>.
+     * Adds a new header with the specified name and value.
      *
-     * @param name the name of the header being added
-     * @param value the value of the header being added
+     * @param name the name of the header
+     * @param value the value of the header
+     * @throws NullPointerException if either {@code name} or {@code value} is null.
      * @return {@code this}
      */
     Headers<T> add(T name, T value);
 
     /**
-     * Adds a new header with the specified name and values. This getMethod can be represented approximately as the
-     * following code:
+     * Adds a new header with the specified name and values. This method is equivalent to
      *
      * <pre>
-     * for (Object v : values) {
-     *     if (v == null) {
-     *         break;
-     *     }
+     * for (T v : values) {
      *     headers.add(name, v);
      * }
      * </pre>
      *
-     * @param name the name of the headepublic abstract rs being set
-     * @param values the values of the headers being set
+     * @param name the header name
+     * @param values the values of the header
      * @return {@code this}
      */
     Headers<T> add(T name, Iterable<? extends T> values);
 
     /**
-     * Adds a new header with the specified name and values. This getMethod can be represented approximately as the
-     * following code:
+     * Adds a new header with the specified name and values. This method is equivalent to
      *
      * <pre>
-     * for (Object v : values) {
-     *     if (v == null) {
-     *         break;
-     *     }
+     * for (T v : values) {
      *     headers.add(name, v);
      * }
      * </pre>
      *
-     * @param name the name of the headepublic abstract rs being set
-     * @param values the values of the headers being set
+     * @param name the header name
+     * @param values the values of the header
      * @return {@code this}
      */
     Headers<T> add(T name, T... values);
 
     /**
-     * Adds a new header with the specified name and value. If the specified value is not a {@link String}, it is
-     * converted into a {@link String} by {@link Object#toString()}, except in the cases of {@link java.util.Date} and
-     * {@link java.util.Calendar}, which are formatted to the date format defined in <a
-     * href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3.1">RFC2616</a>.
+     * Adds a new header. Before the {@code value} is add, it's converted to type {@code T} by a call to
+     * {@link ValueConverter#convertObject(java.lang.Object)}.
      *
-     * @param name the name of the header being added
-     * @param value the value of the header being added
+     * @param name the header name
+     * @param value the value of the header
```

Comment:
The rule is simple (sort of):

- When it's a single clause, it should not end with a period. (a single clause)
- When it's a clause followed by a statement, it should end with a period. (a clause. A statement.)
- When it's a statement, it should end with a period (A statement. Another statement.)

I know we recently put some periods even for single-clause comments, but we need to remove the period for them. I'm sure this could be wrong, though. Please let me know if this rule needs an update.